fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -121,11 +123,16 @@ where
         power_diff(curr_power_table, next_power_table)
     };
 
-    emit!(NewBottomUpCheckpoint {
-        block_height: height.value(),
-        block_hash: &hex::encode(block_hash),
-        num_msgs,
-        next_configuration_number,
+    emit(CheckpointCreated {
+        height: height.value(),
+        hash: HexEncodableBlockHash(block_hash.to_vec()),
+        msg_count: num_msgs,
+        config_number: next_configuration_number,
+    });
+
+    emit(CheckpointFinalized {
```

Comment:
@cryptoAtwill You will probably know this. I am not exactly sure we can say it's finalised at this point right? If not where else would be a good place to emit this event? Any idea please?

Reply:
You can do this in the relayer, once a quorum formed event is detected, it's a good time to emit this event